Apps/WebClient/src/ClientApp/src/store/modules/patientData/getters.ts

```diff
@@ -26,12 +30,35 @@ export const getters: PatientDataGetters = {
     },
     patientData(
         state: PatientDataState
-    ): (hdid: string) => PatientData | undefined {
-        return (hdid: string) => getPatientDataRecordState(state, hdid).data;
+    ): (hdid: string, patientDataTypes: PatientDataType[]) => PatientData[] {
+        return (hdid: string, patientDataTypes: PatientDataType[]) => {
+            const records: PatientData[] = [];
+            const data = getPatientDataRecordState(state, hdid).data;
+            if (data) {
+                patientDataTypes.forEach((patientDataType) => {
+                    const patientData = data[patientDataType];
+                    if (patientData) {
+                        records.push(...patientData);
+                    }
+                });
+            }
+            return records;
+        };
     },
-    isPatientDataLoading(state: PatientDataState): (hdid: string) => boolean {
+    patientDataAreLoading(state: PatientDataState): (hdid: string) => boolean {
         return (hdid: string) =>
             getPatientDataRecordState(state, hdid).status ===
             LoadStatus.REQUESTED;
     },
+    patientDataCount(
+        state: PatientDataState,
+        getters: PatientDataGetters
+    ): (hdid: string, patientDataTypes: PatientDataType[]) => number {
+        return (hdid: string, patientDataTypes: PatientDataType[]) => {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            const data = getters.patientData(hdid, patientDataTypes);
+            return data.length;
+        };
+    },
```

Comment:
```suggestion
    patientDataCount(
        _state: PatientDataState,
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        getters: any
    ): (hdid: string, patientDataTypes: PatientDataType[]) => number {
        return (hdid: string, patientDataTypes: PatientDataType[]) => {
            const data = getters.patientData(hdid, patientDataTypes);
            return data.length;
        };
    },
```

Reply:
lol I do prefer not messing with the ts compiler, good suggestion.